aws-android-sdk-core/src/main/java/com/amazonaws/util/URIBuilder.java

```diff
@@ -0,0 +1,155 @@
+package com.amazonaws.util;
```

Comment:
I assume this is copied over from another module?

Reply:
If that's the case, we should refactor that other module to use this new core capability, and then remove the duplicate.

Reply:
This structure is actually something I wrote up for this because I found it inconvenient to work with `URI` class. Is there an issue with the builder?